File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02952/p02952_num2_parsed.py
def count_odd_digits(N):
    count = 0
    for i in range(1, N+1):
        # Check if the number of digits is odd
        if len(str(i)) % 2 != 0:
            count += 1
    return count

N = int(input())
result = count_odd_digits(N)
print(result)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02952/p02952_num2_parsed.cpp
#include <iostream>
using namespace std;

int countOddDigits(int n) {
    int count = 0;
    for (int i = 1; i <= n; i++) {
        int numDigits = 0;
        int temp = i;
        while (temp > 0) {
            temp /= 10;
            numDigits++;
        }
        if (numDigits % 2 != 0)
            count++;
    }
    return count;
}

int main() {
    int N;
    cin >> N;
    int result = countOddDigits(N);
    cout << result << endl;
    return 0;
}